src/installer/pkg/sfx/installers/osx_scripts/host-x64/postinstall

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+#
+# Licensed to the .NET Foundation under one or more agreements.
+# The .NET Foundation licenses this file to you under the MIT license.
+#
+
+PACKAGE=$1
+INSTALL_DESTINATION=$2
+
+_machine=`uname -m`
+
+# A temporary fix for the permissions issue(s)
+chmod 755 $INSTALL_DESTINATION/dotnet
+
+if ["$_machine" == "amd64"] || ["$_machine" == "x86_x64"]; then
```

Comment:
Given the POSIX shell shebang on line 1 `#!/bin/sh`, `==` (bash/zsh syntax) is not available in strict mode. Also missing spaces is a syntax error in all three shells.
```suggestion
if [ "$_machine" = "amd64" ] || [ "$_machine" = "x86_x64" ]; then
```

Reply:
Thanks, actually refactored this a bit.  I'm a bash noob so I'm learning quite a bit.  Still testing, but could use another review shortly.

Reply:
In PoweShell, `[` is part of the language syntax where as in Unix shells, it's a built-in command with same functionality as `test`. Without the whitespace after `[`, it is iterpreted as `if test"$_machine = "amd64"`.

```sh
# [ is available in POSIX shell and its derivates (bash, ash, dash, zsh etc.)
bash-3.2$ command -v [
[

bash-3.2$ type -a [
[ is a shell builtin
[ is /bin/[

# ---

# [[ is only available in bash, ash, dash, zsh etc.
bash-3.2$ command -v [[
[[

bash-3.2$ type -a [[
[[ is a shell keyword
```